translate/pb/pb_decoder_entry.go

```diff
@@ -0,0 +1,81 @@
+package pb
+
+import (
+	"fmt"
+
+	"github.com/arr-ai/arrai/rel"
+	"github.com/arr-ai/arrai/tools"
+	"google.golang.org/protobuf/proto"
+	"google.golang.org/protobuf/reflect/protoreflect"
+	"google.golang.org/protobuf/types/dynamicpb"
+)
+
+// StdProtobufDecoder transforms the protocol buffer message to a tuple.
+// Sample code to call this method:
+// let sysl = //encoding.proto.decode(//encoding.proto.proto, //os.file('sysl.pb'));
+// let decodeSyslPb = //encoding.proto.decode(sysl);
+// let shop = decodeSyslPb('Module', //os.file("petshop.pb"));
+// petshop.apps("PetShopApi")
+var StdProtobufDecoder = rel.NewNativeFunction("decode", func(tupleParam rel.Value) (rel.Value, error) {
+	tuple, isTuple := tupleParam.(rel.Tuple)
+	if !isTuple {
+		return nil, fmt.Errorf("//encoding.proto.decode: tupleParam not tuple")
+	}
+
+	if !tuple.IsTrue() {
+		// the call looks like `let sysl = //encoding.proto.decode(//encoding.proto.proto, //os.file('sysl.pb'))`
+		return rel.NewNativeFunction("decode$1", func(definition rel.Value) (rel.Value, error) {
+			_, is := tools.ValueAsBytes(definition)
+			if !is {
+				return nil, fmt.Errorf("//encoding.proto.decode: definition not bytes")
+			}
+
+			return rel.NewTuple(rel.NewAttr("fileDescriptor", definition)), nil
```

Comment:
Why do we need this `(fileDescriptor: *)` boilerplate, rather than just returning `definition`? It introduces an extra level of abstraction with the concept of "a structure that contains a fileDescriptor", but I don't see the benefit of doing so.

Reply:
`rel.NewTuple(rel.NewAttr("fileDescriptor", definition))` is a tuple which has `sysl.proto` definition, which can support the syntax:
```
@> /set sysl = //encoding.proto.decode(//encoding.proto.proto, //os.file("sysl.pb"))
@> /set decodeSyslPb = //encoding.proto.decode(sysl)
```
the first argument of `//encoding.proto.decode` is an arr.ai tuple. If it return definition directly, it has to check data type which is `rel.Tuple` or `rel.Bytes` to see now it is the first call with `//encoding.proto.proto` as the first argument or second call.
Now it make sure the first argument is `rel.Tuple` and there is a check:
```
if !isTuple {
		return nil, fmt.Errorf("//encoding.proto.decode: tupleParam not tuple")
	}
```